[zeonchew0324] iP

zeonchew0324 opened the pull request:
# Velma

> "The only way to do great work is to do what you love" - Steve Jobs

Velma helps you manage your tasks easily and efficiently. It's :

-text-based
-simple to use
-easy to manage

Most important of all, it's **free to use !** 

Steps to use Velma:

1. Download it from [here](https://github.com/zeonchew0324/ip).
2. Add your tasks.
3. View all your to-dos on Velma. 😉 
4. Feel free to edit or modify your task list. ✅ 

Features:
- [x] Task management
- [ ] Reminders (coming soon)
- [ ] Event scheduling (coming soon)

Let's check out the main method of Velma:

```java
public static void main(String[] args) {
    new Velma().run();
}

//Enjoy Velma!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

zeonchew0324 committed:
Created initial skeleton of Velma chatbot

zeonchew0324 committed:
Added horizontal lines

zeonchew0324 committed:
chatbot echoes command from user

zeonchew0324 committed:
check if request is bye to exit

zeonchew0324 committed:
added request into string array and able to list back requests

zeonchew0324 committed:
added horizontal lines when list is called

zeonchew0324 committed:
Created task class

zeonchew0324 committed:
changed list representation

zeonchew0324 committed:
Created mark ability

zeonchew0324 committed:
created unmark ability

zeonchew0324 committed:
changed coding representation

zeonchew0324 committed:
added three new Task subclass

zeonchew0324 committed:
added todo request

zeonchew0324 committed:
added request for deadline

zeonchew0324 committed:
added request for event

zeonchew0324 committed:
tested using text-ui-test

zeonchew0324 committed:
created exception class for velma

zeonchew0324 committed:
added other exception case

zeonchew0324 committed:
added delete command

zeonchew0324 committed:
added command enum

zeonchew0324 committed:
added switch case for command enums

zeonchew0324 committed:
Add load tasks method to load the file

zeonchew0324 committed:
Add saveTask method to save task straight into hard disk

zeonchew0324 committed:
Modify main task to save and load task

zeonchew0324 committed:
Add toFileFormat method so that the output is formatted

zeonchew0324 committed:
Create file so that the directory exist to save task

zeonchew0324 committed:
remove the tofileformat so that the tostring representation stays consistent with previous levels

zeonchew0324 committed:
fix indentation

zeonchew0324 committed:
fix indentation

zeonchew0324 committed:
empty

zeonchew0324 committed:
Add localdatetime to ensure that date time format is readable by the chatbot

zeonchew0324 committed:
List command now can specify date and will return tasks ending on that specified date

zeonchew0324 committed:
Add exception such that deadline time must be in correct time format

zeonchew0324 committed:
Change some code styling

zeonchew0324 committed:
Create new classes that refactor for OOP

zeonchew0324 committed:
Change Velma class body to incorporate new classes

zeonchew0324 committed:
Finish integrating the other classes to make code more OOP

zeonchew0324 committed:
Remove unnecessary method

zeonchew0324 committed:
Add javadoc header for each method

zeonchew0324 committed:
add package for each of the files

zeonchew0324 committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

* commit '81a9c537afe8492675eb253968939364d65db091':
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

zeonchew0324 committed:
Create test file for basic junit test

zeonchew0324 committed:
Create jar file using gradle

zeonchew0324 committed:
Create find command to chatbot to find matching task

zeonchew0324 committed:
Fix coding standard

zeonchew0324 committed:
add checkstyle settings

gladwintan commented on src/test/TaskTest.java:
```diff
@@ -0,0 +1,21 @@
+public class TaskTest {
+
+    @Test
+    public void testChangeIsDone() {
```
Could the test be better named according to the conventions?

zeonchew0324 replied:
Yes the test name could have been named better.

gladwintan commented on src/main/java/Velma/task/Task.java:
```diff
@@ -0,0 +1,52 @@
+     * Returns the status of the task.
+     * @return
+     */
+    public boolean getIsDone() {
```
Could the function be renamed to isDone() to shorten it?

gladwintan commented on src/main/java/Velma/task/Task.java:
```diff
@@ -0,0 +1,52 @@
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Good naming for boolean

gladwintan reviewed (commented):
Overall, I found your code to easy to read. Most variables and functions are well-named and intuitive.

zeonchew0324 committed:
Fixing styling error

zeonchew0324 committed:
Change style settings

zeonchew0324 committed:
Finish changing style errors

zeonchew0324 committed:
change javadoc representation

zeonchew0324 committed:
add print line method

zeonchew0324 committed:
Create GUI for Velma

zeonchew0324 committed:
Interact with user

zeonchew0324 committed:
Fix dialog box styling

zeonchew0324 committed:
Using FXML

zeonchew0324 committed:
Add Velma logic into FXML

zeonchew0324 committed:
Customise JavaFX CSS (until borders)

Timenikhil commented on Jar/src/velma/Storage.java:
```diff
@@ -0,0 +1,115 @@
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+import velma.task.Task;
```
I like that you have used lowercase for package names as per the coding standard

Timenikhil commented on Jar/src/velma/task/Task.java:
```diff
@@ -0,0 +1,52 @@
+package velma.task;
+
+public class Task {
+    protected String description;
```
Its nice to see that you have clearly defined the accessibility of the variables and used clear variable names.

Timenikhil commented on Jar/src/velma/task/Task.java:
```diff
@@ -0,0 +1,52 @@
+package velma.task;
+
+public class Task {
```
Is there a reason you chose to go for a concrete class instead of an abstract class?

Timenikhil commented on src/main/java/velma/Storage.java:
```diff
@@ -0,0 +1,126 @@
+    }
+
+
+    /**
```
Is there a reason your javadoc is not aligned with the /*?

Timenikhil commented on src/main/java/velma/task/Todo.java:
```diff
@@ -0,0 +1,17 @@
+ */
+public class Todo extends Task {
+
+    public Todo(String description) {
```
I like that you have followed the indentation standard

Timenikhil commented on src/main/java/velma/Velma.java:
```diff
@@ -0,0 +1,231 @@
+            Command command = getCommand(request);
+
+            try {
+                switch (command) {
```
I like your use of enums and following the switch case styling

Timenikhil commented on src/main/java/velma/Velma.java:
```diff
@@ -0,0 +1,231 @@
+                    }
+                    break;
+
+                case MARK:
```
Is there a reason you skipped adding a  fall through comment here?

Timenikhil reviewed (commented):
Overall great coding standard. Scope for some improvement

zeonchew0324 committed:
Add background image and prompt text

zeonchew0324 committed:
add send message sound

zeonchew0324 committed:
Apply varargs

zeonchew0324 committed:
Add assertions into certain methods.

Assertions can help catch bugs by verifying that certain conditions hold true. For example, using assertion in the findTask method ensures that the method always returns a non-null task.

zeonchew0324 committed:
Use better variable naming to improve readability of the code

zeonchew0324 committed:
Create after task type which the task is specified when to be done after an event

zeonchew0324 committed:
Merge pull request #1 from zeonchew0324/branch-A-Assertions

Add assertions into certain methods.

zeonchew0324 committed:
Merge pull request #2 from zeonchew0324/branch-A-CodeQuality

Use better variable naming to improve readability of the code

zeonchew0324 committed:
Merge pull request #3 from zeonchew0324/branch-B-doAfterTasks

Create after task type which the task is specified when to be done af…

zeonchew0324 committed:
Handle program exit when 'bye' message is sent

zeonchew0324 committed:
Set stage title to Velma

- Updated the stage title in UI to display the chatbot's name, "Velma" in the window.

zeonchew0324 committed:
Add chatbot screenshot into docs

- Added a screenshot of the Velma chatbot with its working features into the docs folder for documentation purpose

zeonchew0324 committed:
Add user guide to README

- Wrote a detailed user guide in the README file to help users understand how to use Velma
- Included sections on the features and event types.

zeonchew0324 committed:
Add welcome message when the chatbot is initialize.

-Velma will now greet the user when the chatbot is launched. This could let user know that Velma is ready to help.

zeonchew0324 committed:
Resolve gradle run error

- Change build gradle file to the correct package name. ClassNotFound exception issue when running gradle is solved

zeonchew0324 committed:
Resolve gradle run error

- Change build gradle file to the correct package name. ClassNotFound exception issue when running gradle is solved

zeonchew0324 committed:
Fix javaFX error

zeonchew0324 committed:
Change styling error for storage class method

The original comment is put at the middle of the class,
this comment is changed to include at before the class.

zeonchew0324 committed:
Fix styling error for dialog box

Missing javadoc header for dialogbox class,
added javadoc header to describe about Velma's dialogbox.

zeonchew0324 committed:
Amend styling error for Launcher class

Added a Javadoc header to describe about launcher class

zeonchew0324 committed:
Fix styling error for after task class

Added a missing Javadoc header to specify after task.

zeonchew0324 committed:
Fix all stying issue after running checkstyle main

Added missing Javadoc headers
and added precedented whitespaces.

zeonchew0324 committed:
Fix all checkstyle error

zeonchew0324 committed:
Fix master error

zeonchew0324 committed:
Resolve code to not have deep nesting

zeonchew0324 committed:
Fix mark command bug

zeonchew0324 committed:
Fix event start time can be after end time bug

zeonchew0324 committed:
Fix mark and delete command index out of bound bug

zeonchew0324 committed:
Fix exception description

zeonchew0324 committed:
Fix missing javadoc comments

zeonchew0324 committed:
Make load method shorter

zeonchew0324 committed:
Fix checkstyle error

zeonchew0324 committed:
Add description

zeonchew0324 committed:
Update README.md
